whylog/constraints/__init__.py

```diff
@@ -0,0 +1,36 @@
+class Constraint(object):
```

Comment:
I am not sure yet, if AbstractConstraint should have any check() function. I was thinking about another abstract method - saving constraint to rulesbase's Rule object.